test/jdk/java/util/jar/JarFile/VerifySignedJar.java

```diff
@@ -41,16 +56,24 @@ private static void Unreached (Object o)
     }
 
     public static void main(String[] args) throws Exception {
-        File f = new File(System.getProperty("test.src", "."), "thawjar.jar");
-        JarFile jf = new JarFile(f);
-        try {
-            // Read entries via Enumeration
-            for (Enumeration e = jf.entries(); e.hasMoreElements();)
-                jf.getInputStream((ZipEntry) e.nextElement());
+
+        Path j = createJar();
+        Path s = signJar(j, keyEntry("cn=duke"));
+
+        try (JarFile jf = new JarFile(s.toFile())) {
+
+            for (JarEntry e: Collections.list(jf.entries())) {
+                // Reading entry to trigger verification
+                jf.getInputStream(e).transferTo(OutputStream.nullOutputStream());
+                // Check that all regular files are signed by duke
+                if (!e.getName().startsWith("META-INF/")) {
+                    checkSignedBy(e, "cn=duke");
+                }
+            }
 
             // Read entry by name
-            ZipEntry ze = jf.getEntry("getprop.class");
-            JarEntry je = jf.getJarEntry("getprop.class");
+            ZipEntry ze = Objects.requireNonNull(jf.getEntry("getprop.class"));
+            JarEntry je = Objects.requireNonNull(jf.getJarEntry("getprop.class"));
```

Comment:
I've removed the local variables. Just in case there was some intent in showing that the methods excercise both ZipFile and JarFile, I updated the comment above slightly.

I also replaced the exception checking code as suggested. I opted for a static import, is that ok?